Sources/Basics/SandboxProfile.swift

```diff
@@ -0,0 +1,108 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 - 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Foundation
+import TSCBasic
+
+/// A sandbox profile representing the desired restrictions. The implementation can vary between platforms.
+public struct SandboxProfile {
+    /// An ordered list of path rules, where the last rule to cover a particular path "wins". These will be resolved
+    /// to absolute paths at the time the profile is applied. They are applied after any of the implicit directories
+    /// referenced by other sandbox profile settings.
+    public var pathAccessRules: [PathAccessRule]
+
+    /// Represents a rule for access to a path and everything under it.
+    public enum PathAccessRule: Equatable {
+        case noaccess(AbsolutePath)
+        case readonly(AbsolutePath)
+        case writable(AbsolutePath)
+    }
+
+    /// Whether to allow outbound and inbound network access.
+    public var allowNetwork: Bool
+
+    /// Configures a SandboxProfile for blocking network access and writing to the file system except to specifically
+    /// permitted locations.
+    public init(_ pathAccessRules: [PathAccessRule] = [], allowNetwork: Bool = false) {
+        self.pathAccessRules = pathAccessRules
+        self.allowNetwork = allowNetwork
+    }
+
+    // Convenience initializer to make it easier to construct sandbox profiles with conditional rules.
+    public init(_ pathAccessRules: PathAccessRule?..., allowNetwork: Bool = false) {
```

Comment:
I'm not 100% sure about this one.  I think I will probably want to remove this and instead have call sites modify the `pathAccessRules` property directly, but am curious about opinions.  This convenience initializer does make call sites a bit more succinct.

Reply:
I'm going to go ahead and remove this, and keep SandboxProfile simple even if the clients get a little more complicated.

Reply:
Resolved this.